SU2_CFD/include/output/filewriter/CSurfaceFEMDataSorter.hpp

```diff
@@ -74,7 +75,34 @@ class CSurfaceFEMDataSorter final: public CParallelDataSorter{
   unsigned long GetGlobalIndex(unsigned long iPoint) const override {
     return globalSurfaceDOFIDs[iPoint];
   }
+  
+  /*!
+    * \brief Get the beginning global renumbered node ID of the linear partition owned by a specific processor.
+    * \param[in] rank - the processor rank.
+    * \return The beginning global renumbered node ID.
+    */
+   unsigned long GetNodeBegin(unsigned short rank) const override {
+     unsigned long offsetSurfaceDOFs = 0;
+     for(int i=0; i<rank; ++i) offsetSurfaceDOFs += nSurfaceDOFsRanks[i];
```

Comment:
Make this unsigned short? Or does the compiler optimize this anyway?
btw. unsigned short is max= 65,535 ... that would be quite a lot of processors but probably not too far away from possible for some people out there (idk)?

Reply:
Unsigned shorts are nonsense... Unless the purpose is to reduce storage and/or vectorize calculations.

Reply:
`grep -r "unsigned short" SU2_CFD/ Common/ | wc -l` ->10564
It is used all over the place in SU2. 
I found Stroustroup, Sutter & Carruth [here (~2min)](https://www.youtube.com/watch?v=Puio5dly9N8&t=2560) with their bits on the topic. I guess these are the reasons you mean as well

Reply:
They take longer to write too xD.
We should define su2int as either int or long and do a replace all :smiling_imp: